src/allmydata/test/test_abbreviate.py

```diff
@@ -0,0 +1,147 @@
+"""
+Tests for allmydata.util.abbreviate.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future.utils import PY2
+if PY2:
+    from builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, int, list, object, range, str, max, min  # noqa: F401
+
+from datetime import timedelta
+
+from twisted.trial import unittest
+
+from allmydata.util import abbreviate
+
+
+class Abbreviate(unittest.TestCase):
+    def test_abbrev_time_1s(self):
+        diff = timedelta(seconds=1)
+        s = abbreviate.abbreviate_time(diff)
+        self.assertEqual('1 second ago', s)
```

Comment:
GitHub won't let me comment on the unmodified `abbreviate_time` implementation so I'll comment here...  `abbreviate_time` is ported such that it returns a "native string" (native strings in these tests giving one hint about this).  `abbreviate_time` had no API docs before but maybe now it should at least document this fact to make life as easy as possible when interacting with it in further porting work?

Also I wonder if there is a general rule or set of rules that can be used to decide when "native strings" are appropriate.  I'm guessing ... not.  And everything is just a pile of special cases to consider.  How did you conclude native strings were the right way to go here?

Reply:
There's a unicode_literals future import, so it'll be Unicode?

I decided on unicode cause it's definitely human readable text, rather than something a computer would read. Like, if the answer to "would I internationalize this" is yes it should definitely be unicode.

Reply:
Added docstring to two main APIs.

Reply:
Oh right!  Sorry, I forgot.  I'll try to remember that there *are* no native literals in a ported module anymore.  Thanks.